### PR TITLE
[16.0][mig] purchase_order_no_zero_price: Migration to 16.0

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

**Comment by PavarunCh** [comment]
I noticed that there are a lot of commits, Please squash commits.

**Comment by dsolanki-initos** [comment]
> I noticed that there are a lot of commits, Please squash commits.

@PavarunCh, I did only two commits (pre-commit and migration), the remaining are coming from the patch of version 15.

**Comment by PavarunCh** [comment]
I apologize for the misunderstanding.

**Comment by dsolanki-initos** [comment]
> I apologize for the misunderstanding.

Okay, No worries, Could you please review this PR?

**Review by PavarunCh — APPROVED** [review]
LGTM :+1: 
Functional tested

**Review by HviorForgeFlow — APPROVED** [review]
Code review LGTM :+1:

File: purchase_order_no_zero_price/models/purchase_order_line.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.constrains("price_unit", "display_type", "state")
+    def _check_price_unit_zero(self):
+        # Prevent zero prices on confirmed POs
+        # to help ensure proper Stock Valuation
+        # when receiving the products
+        # Also supports `purchase_triple_discount` module
+        no_price_lines = self.filtered(
```
Comment (rousseldenis) [inline purchase_order_no_zero_price/models/purchase_order_line.py:16]: @HviorForgeFlow @dsolanki-initos IMHO this should be configurable either on the company level, either on product one as it breaks standard behavior.

Modules that are not aware of this one (no dependency) could fail.

**Pull request merged**